src/mlpack/core/data/scaler_methods/pcawhitening.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file pcawhitening.hpp
+ * @author Jeffin Sam
+ *
+ * Whitening scaling to scale features, Using PCA Whitening.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_PCA_WHITENING_SCALE_HPP
+#define MLPACK_CORE_DATA_PCA_WHITENING_SCALE_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/lin_alg.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple PcaWhitening class.
+ *
+ * Whitens a matrix using the eigendecomposition of the covariance matrix.
+ * Whitening means the covariance matrix of the result is the identity matrix.
+ * 
+ * For whitening related formula and more info, check the link below.
+ * http://ufldl.stanford.edu/tutorial/unsupervised/PCAWhitening/
+ *
+ * @code
+ * arma::mat input;
+ * Load("train.csv", input);
+ * arma::mat output;
+ *
+ * // Fit the features.
+ * PcaWhitening scale;
+ * scale.Fit(input)
+ *
+ * // Scale the features.
+ * scale.Transform(input, output);
+ *
+ * // Retransform the input.
+ * scale.InverseTransform(output, input);
+ * @endcode
+ */
+class PcaWhitening
+{
+ public:
+  /**
+  * A constructor to set the regulatization parameter.
+  *
+  * @param eps Regularization parameter.
+  */
+  PcaWhitening(double eps = 0.00005)
+  {
+    epsilon = eps;
+  }
+
+  /**
+  * Function to fit features, to find out the min max and scale.
+  *
+  * @param input Dataset to fit.
+  */
+  template<typename MatType>
+  void Fit(const MatType& input)
+  {
+    itemMean = arma::mean(input, 1);
+    // Get eigenvectors and eigenvalues of covariance of input matrix.
+    eig_sym(eigenValues, eigenVectors, mlpack::math::ColumnCovariance(
+        input.each_col() - itemMean));
+    for (size_t i = 0; i < eigenValues.n_elem; i++)
+        eigenValues(i) = eigenValues(i) + epsilon;
```

Comment:
Actually you can avoid the loop here and just write `eigenValues += epsilon`.

Reply:
Thanks for this :)